### PR TITLE
chore: reorganize StorageManager files

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
* Move useStateManager hook's reducer and actions to separate files
* Move useDropZone to hooks folder
* Copy uploadFile to StorageManager utils to make it easier to do split packages later (leaving existing one for soon-deprecated FileUploader)

#### Description of how you validated changes
yarn next-example dev

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Storage/StorageManager/hooks/useUploadFiles/useUploadFiles.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export function useUploadFiles({
             progressCallback: onProgress,
             errorCallback: onError,
             provider,
-          }) as unknown as UploadTask; // @todo: fix type returned from uploadFile
+          }) as unknown as UploadTask;
```
Comment (ioanabrooks) [inline packages/react/src/components/Storage/StorageManager/hooks/useUploadFiles/useUploadFiles.ts:88]: Do we still need the cast to unknown here?

**Review by ioanabrooks — COMMENTED** [review]
✨ No unit tests though 💀

File: packages/react/src/components/Storage/StorageManager/hooks/useUploadFiles/useUploadFiles.ts (hunk 1 above)
Comment (reesscot) [reply packages/react/src/components/Storage/StorageManager/hooks/useUploadFiles/useUploadFiles.ts:88]: Unfortunately, I couldn't get uploadFile to return UploadTask due to how Amplify JS Storage types work...

**Comment by reesscot** [comment]
> ✨ No unit tests though 💀

Unit tests to follow

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**